assets/reader-activation/auth.js

```diff
@@ -54,299 +54,307 @@ const convertFormDataToObject = ( formData, includedFields = [] ) =>
-			ev.preventDefault();
-			container.classList.remove( 'newspack-reader__auth-form__visible' );
-			container.style.display = 'none';
-			displayCurrentlyOpenOverlayPrompts();
-		} );
+			const actionInput = form.querySelector( 'input[name="action"]' );
+			const emailInput = form.querySelector( 'input[name="email"]' );
+			const passwordInput = form.querySelector( 'input[name="password"]' );
+			const redirectInput = form.querySelector( 'input[name="redirect"]' );
+			const submitButtons = form.querySelectorAll( '[type="submit"]' );
+			const closeButton = container.querySelector( 'button[data-close]' );
 
-		const messageContentElement = container.querySelector(
-			'.newspack-reader__auth-form__response__content'
-		);
+			if ( closeButton ) {
+				closeButton.addEventListener( 'click', function ( ev ) {
+					ev.preventDefault();
+					container.classList.remove( 'newspack-reader__auth-form__visible' );
+					container.style.display = 'none';
+					displayCurrentlyOpenOverlayPrompts();
+				} );
+			}
 
-		const authLinkMessage = container.querySelector( '[data-has-auth-link]' );
-		authLinkMessage.hidden = true;
+			const messageContentElement = container.querySelector(
+				'.newspack-reader__auth-form__response__content'
+			);
 
-		let accountLinks, triggerLinks;
-		const initLinks = function () {
-			accountLinks = document.querySelectorAll( '.newspack-reader__account-link' );
-			triggerLinks = document.querySelectorAll( '[data-newspack-reader-account-link]' );
-			triggerLinks.forEach( link => {
-				link.addEventListener( 'click', handleAccountLinkClick );
-			} );
-		};
-		initLinks();
-		/** Re-initialize links in case the navigation DOM was modified by a third-party. */
-		setTimeout( initLinks, 1000 );
+			const authLinkMessage = container.querySelector( '[data-has-auth-link]' );
+			authLinkMessage.hidden = true;
 
-		/**
-		 * Handle reader changes.
-		 */
-		function handleReaderChanges() {
-			const reader = readerActivation.getReader();
-			emailInput.value = reader?.email || '';
-			if ( accountLinks?.length ) {
-				accountLinks.forEach( link => {
-					/** If there's a pre-auth, signing in redirects to the reader account. */
-					if ( reader?.email && ! reader?.authenticated ) {
-						link.setAttribute( 'data-redirect', link.getAttribute( 'href' ) );
-					} else {
-						link.removeAttribute( 'data-redirect' );
-					}
-					try {
-						const labels = JSON.parse( link.getAttribute( 'data-labels' ) );
-						link.querySelector( '.newspack-reader__account-link__label' ).textContent =
-							reader?.email ? labels.signedin : labels.signedout;
-					} catch {}
+			let accountLinks, triggerLinks;
+			const initLinks = function () {
+				accountLinks = document.querySelectorAll( '.newspack-reader__account-link' );
+				triggerLinks = document.querySelectorAll( '[data-newspack-reader-account-link]' );
+				triggerLinks.forEach( link => {
+					link.addEventListener( 'click', handleAccountLinkClick );
```

Comment:
This too, and the `handleAccountLinkClick()` and `handleReaderChanges()` functions.

Reply:
`initLinks` can live outside the loop, but both `handleAccountLinkClick` and `handleReaderChanges` functions rely on form fields inside the specific containers, no?

Reply:
Oops! You're right, they mix account link and form actions.

Reply:
Ah, but they do need to get moved outside the loop (and then create their own internal loops to handle multiple forms) because otherwise the account links will only trigger the last instance of the form on the page. This will probably be the modal form, but b08d5bc refactors to ensure that all form instances are handled.